src/ImageData/FileLoader.cc

```diff
@@ -65,28 +65,80 @@ FileLoader* FileLoader::GetLoader(std::shared_ptr<casacore::ImageInterface<float
     }
 }
 
-FileLoader::FileLoader(const std::string& filename) : _filename(filename) {}
+FileLoader::FileLoader(const std::string& filename, bool is_gz)
+    : _filename(filename), _is_gz(is_gz), _modify_time(0), _num_dims(0), _has_pixel_mask(false) {
+    // Set initial modify time
+    ImageUpdated();
+}
 
 bool FileLoader::CanOpenFile(std::string& /*error*/) {
     return true;
 }
 
-bool FileLoader::GetShape(IPos& shape) {
-    ImageRef image = GetImage();
-    if (image) {
-        shape = image->shape();
-        return true;
+typename FileLoader::ImageRef FileLoader::GetImage() {
+    if (!_image) {
+        OpenFile(_hdu);
+    }
+
+    return _image;
+}
+
+void FileLoader::CloseImageIfUpdated() {
+    // Close image if updated when only the loader owns it unless decompressed
+    if (!_is_gz && _image.unique() && ImageUpdated()) {
+        _image->tempClose();
+    }
+}
+
+bool FileLoader::ImageUpdated() {
+    bool changed(false);
+    casacore::File ccfile(_filename);
+    auto updated_time = ccfile.modifyTime();
```

Comment:
at some stage I'd like us to move all our filesystem stuff to `std::filesystem` instead of `casacore::File`, because I think we should be sticking with STL variants of things where they exist. But that's a separate issue for another time :smile:

Reply:
The return value of std::filesystem::last_write_time is std::chrono::time_point.  It is non-trivial (if not impossible) to convert this to std::time_t in a portable way until we use C++20.